src/test/java/io/supertokens/test/CronjobTest.java

```diff
@@ -699,4 +700,80 @@ public void testPerUserPoolCronTask() throws Exception {
         process.kill();
         assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
     }
+
+    @Test
+    public void testThatCoreAutomaticallySyncsToConfigChangesInDb() throws Exception {
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args, false);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{EE_FEATURES.MULTI_TENANCY});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        if (StorageLayer.isInMemDb(process.getProcess())) {
+            return;
+        }
+
+        TenantIdentifier t1 = new TenantIdentifier(null, "a1", null);
+        Multitenancy.addNewOrUpdateAppOrTenant(process.getProcess(), new TenantConfig(
+                t1,
+                new EmailPasswordConfig(false),
+                new ThirdPartyConfig(false, null),
+                new PasswordlessConfig(false),
+                new JsonObject()
+        ), false);
+
+        boolean found = false;
+
+        TenantConfig[] allTenants = MultitenancyHelper.getInstance(process.getProcess()).getAllTenants();
+        for (TenantConfig tenant : allTenants) {
+            if (tenant.tenantIdentifier.equals(t1)) {
+                assertFalse(tenant.emailPasswordConfig.enabled);
+                found = true;
+            }
+        }
+        assertTrue(found);
+
+        MultitenancyStorage storage = (MultitenancyStorage) StorageLayer.getStorage(process.getProcess());
+        storage.overwriteTenantConfig(new TenantConfig(
+                t1,
+                new EmailPasswordConfig(true),
+                new ThirdPartyConfig(false, null),
+                new PasswordlessConfig(false),
+                new JsonObject()
+        ));
+
+        // Check that it was not updated in memory yet
+        found = false;
+        allTenants = MultitenancyHelper.getInstance(process.getProcess()).getAllTenants();
+        for (TenantConfig tenant : allTenants) {
+            if (tenant.tenantIdentifier.equals(t1)) {
+                assertFalse(tenant.emailPasswordConfig.enabled);
+                found = true;
+            }
+        }
+        assertTrue(found);
+
+        // Wait for the cronjob to run
+        Thread.sleep(61000);
```

Comment:
im not sure this is a good idea to wait for 1 min in a test.. we should instead change the cronjob time to run every 10 seconds (see how it's done in other cronjobs) and wait for 12 seconds or something